### PR TITLE
feat: add `sum` function

### LINKED ISSUE #76: Add `sum` function
# Suggestion

## 💻 Use Cases
```ts
sum([1,2,3,4,5]); // 15
sum(['a', 'b', 'c']); // 'abc'

await sum(toAsync([1,2,3,4,5])); // 15
await sum(toAsync(['a', 'b', 'c'])); // 'abc'
```

### PR DESCRIPTION
**Pull request by ppeeou**
Fixes #76

### CODE REVIEW

File: src/sum.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * ```ts
+ * sum([1, 2, 3, 4]); // 10
+ * sum(['a', 'b', 'c']); // 'abc'
+ * await sum(toAsync([1, 2, 3, 4])); // 10
+ * await sum(toAsync(['a', 'b', 'c'])); // 'abc'
+ * ```
+ */
+function sum<
+  A extends Iterable<number | string> | AsyncIterable<number | string>,
+>(iterable: A): ReturnSumType<A>;
```
Comment (shine1594) [inline src/sum.ts:24]: This code seems to be unnecessary.

File: src/sum.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import add from "./add";
+import reduce from "./reduce";
+import IterableInfer from "./types/IterableInfer";
+import ReturnValueType from "./types/ReturnValueType";
+import { isAsyncIterable, isIterable } from "./_internal/utils";
+
+type ReturnSumType<
+  A extends Iterable<number | string> | AsyncIterable<number | string>,
+> = ReturnValueType<A, IterableInfer<A> extends string ? string : number>;
```
Comment (shine1594) [inline src/sum.ts:9]:
```suggestion
> = ReturnValueType<A, IterableInfer<A>>;
```

Comment (ppeeou) [reply src/sum.ts]: It's look better👍

**Review by shine1594 — APPROVED** [review]
👍

**Pull request merged**